### ISSUE TITLE
System.InvalidOperationException: hsmKeyIdentifier

### ISSUE THREAD

**Issue by MrM1y4g1** [thycotic-ps/thycotic.secretserver#384]
### Verified issue does not already exist?

Yes

### What error did you receive

System.InvalidOperationException: The hsmKeyIdentifier property was not found for the Thycotic.PowerShell.Secrets.Secret object.

### Provide a test case or steps to reproduce

Perform any action that uses Get-TssSecret function.

### Expected behavior

Returned credentials :)

### What Edition of Secret Server?

Cloud Subscription

### What version of Secret Server

Secret Server Cloud

### What PowerShell host was used when producing this error

PowerShell Core (pwsh)

### PowerShell Host Version

Name                           Value
----                           -----
PSVersion                      7.3.10
PSEdition                      Core
GitCommitId                    7.3.10
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0ΓÇª}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

**Comment by MrM1y4g1** [comment]
There's a quick work around for now by editing this line:
https://github.com/thycotic-ps/thycotic.secretserver/blob/20479d55cb7c46eb3a156ed5b63f09484ae84fcd/src/functions/secrets/Get-TssSecret.ps1#L240

To look as such:
```ps
if ($restResponse) {
    [Thycotic.PowerShell.Secrets.Secret]($restResponse | Select-Object -ExcludeProperty 'hsmKeyIdentifier')
}
```

**Comment by jagger** [comment]
Excluding properties works, but limits functionality. 

Properties need to be updated on the secret class to address this.

**Comment by MrM1y4g1** [comment]
Yep, that would definately be the proper fix, though from what i'm seeing the field is empty as is right now so excluding it dosen't change much. though it'll need a proper fix in the class file as you said :)

**Comment by jagger** [comment]
Class update included in PR  #386
